index.js

```diff
@@ -44,4 +39,45 @@ const nextOnNetlify = (options = {}) => {
   trackNextOnNetlifyFiles();
 };
 
+const watch = (functionsPath, publishPath) => {
+  logTitle(`👀 Watching source code for changes`);
+
+  const watcher = chokidar.watch(SRC_FILES);
+  const runBuild = debounceFn(
+    () => {
+      try {
+        execa.sync("next", ["build"], { stdio: "inherit" });
+        build(functionsPath, publishPath);
+      } catch (e) {
+        console.log(e);
+      }
+    },
+    {
+      wait: 3000,
```

Comment:
Would this be a problem if `next build` takes more than 3 seconds?
Would it be better to debounce based on "is next build running?" instead of a duration?

Reply:
To sum up an internal discussion we had about this.
`next build` calls cannot interleave since each `chokidar` event is scheduled to run after the current `next build` is finished.

If/when we [change to async I/O](https://github.com/netlify/next-on-netlify/issues/86) we need to re-visit this approach.